File: foliage-sample/src/main/java/com/github/lavenderx/rpc/sample/server/RpcBootstrap.java
package com.github.lavenderx.rpc.sample.server;

import com.github.lavenderx.rpc.sample.config.ServerConfig;
import org.springframework.context.annotation.AnnotationConfigApplicationContext;
import org.springframework.context.support.AbstractApplicationContext;

public class RpcBootstrap {

    public static void main(String[] args) {
        final AbstractApplicationContext ctx =
                new AnnotationConfigApplicationContext(ServerConfig.class);

        Runtime.getRuntime().addShutdownHook(new Thread(ctx::close));
    }
}

File: foliage-sample/src/main/kotlin/io/foliage/netty/rpc/sample/RpcBootstrap.kt
package io.foliage.netty.rpc.sample

import io.foliage.netty.rpc.sample.config.ServerConfig
import org.springframework.context.annotation.AnnotationConfigApplicationContext

class RpcBootstrap {
    companion object {
        @JvmStatic fun main(args: Array<String>) {
            val ctx = AnnotationConfigApplicationContext(ServerConfig::class.java)

            Runtime.getRuntime().addShutdownHook(Thread(Runnable { ctx.close() }))
        }
    }
}